### PR TITLE
feat: add code block language select

### PR DESCRIPTION
**Pull request by marcoSven**
### Description

A floating language select dropdown for the code block extension

### Checklist

- [x] I have read the [**contributing**](https://github.com/remirror/remirror/blob/HEAD/docs/contributing.md) document.
- [x] My code follows the code style of this project and `pnpm fix` completed successfully.
- [ ] I have updated the documentation where necessary.
- [ ] New code is unit tested and all current tests pass when running `pnpm test`.

### Screenshots

Screenshot 2023-07-11 at 08 58 42

https://github.com/remirror/remirror/assets/3860298/9197e0eb-d5c9-49e7-911e-5a5cfc7d5c35

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -3294,10 +3349,10 @@ importers:
         version: 7.0.18(react-dom@18.2.0)(react@18.2.0)
       '@storybook/builder-vite':
         specifier: ^7.0.18
-        version: 7.0.18(typescript@5.1.6)(vite@4.3.9)
+        version: 7.0.18(typescript@5.0.4)(vite@4.3.9)
       '@storybook/builder-webpack5':
         specifier: ^7.0.18
-        version: 7.0.18(@swc/core@1.3.60)(esbuild@0.17.19)(react-dom@18.2.0)(react@18.2.0)(typescript@5.1.6)
+        version: 7.0.18(@swc/core@1.3.60)(esbuild@0.17.19)(react-dom@18.2.0)(react@18.2.0)(typescript@5.0.4)
```
Comment (marcoSven) [inline pnpm-lock.yaml:3355]: Noticed this downgrade

File: packages/remirror__extension-react-code-block-language-select/package.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
+{
+  "name": "@remirror/extension-react-code-block-language-select",
```
Comment (ocavue) [inline packages/remirror__extension-react-code-block-language-select/package.json:2]: I feel this package name is too long. Maybe just `@remirror/extension-react-language-select`? Up to you.

File: packages/remirror__extension-react-code-block-language-select/readme.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+## Usage
+
+The following code creates an instance of this extension.
+
+```ts
+import { CodeBlockLanguageSelect } from '@remirror/extension-react-code-block-language-select';
+
+const extension = new CodeBlockLanguageSelect();
+```
```
Comment (ocavue) [inline packages/remirror__extension-react-code-block-language-select/readme.md:36]: I guess this part should be removed

File: packages/remirror__theme/src/extension-code-block-theme.ts [hunk 4]
```diff
@@ -7,6 +7,22 @@ export const WRAP = css`
   white-space: pre-wrap !important;
 `;
 
+export const LANGUAGE_SELECT = css`
+  pre[class*='language-'] {
+    padding: 2em 1em !important;
```
Comment (ocavue) [inline packages/remirror__theme/src/extension-code-block-theme.ts:12]: Why do we need `important` here? Is there a better way, like increasing the selector level?

**Comment by ocavue** [comment]
Very impressive! I will test it more tomorrow since it's quite late here.

**Comment by marcoSven** [comment]
Thank you @ocavue, I think I was able to address all your feedback.

Below is a recording of the story I added for the new component.

https://github.com/remirror/remirror/assets/3860298/a6589d16-6d26-4569-83cb-87696297aad1

File: packages/remirror__extension-react-code-block-language-select/package.json [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  "version": "1.0.0",
```
[the remark this answers is not in the document]
Comment (marcoSven) [reply packages/remirror__extension-react-code-block-language-select/package.json]: I missed your last comment. Do we need to change `0.1.0` to `0.0.0`?

File: packages/remirror__extension-code-block/src/code-block-utils.ts [hunk 6]
```diff
@@ -244,12 +244,10 @@ export function getLanguage(props: GetLanguageProps): string {
...
 export function codeBlockToDOM(node: ProsemirrorNode, extra: ApplySchemaAttributes): DOMOutputSpec {
   const { language, wrap } = omitExtraAttributes(node.attrs, extra);
-  const { style: _, ...extraAttrs } = extra.dom(node);
+  const { ...extraAttrs } = extra.dom(node);
   let style = extraAttrs.style;
 
-  if (wrap) {
-    style = joinStyles({ whiteSpace: 'pre-wrap', wordBreak: 'break-all' }, style);
-  }
+  style = wrap ? joinStyles({ whiteSpace: 'pre-wrap', wordBreak: 'break-all' }, style) : style;
```
Comment (ocavue) [inline packages/remirror__extension-code-block/src/code-block-utils.ts:250]: The style will be added to both the `pre` element and the `code` element. This cause some weird behavior (e.g. The first line of the code block has a padding but other lines don't)
CleanShot 2023-07-15 at 00 56 21@2x

Comment (ocavue) [reply packages/remirror__extension-code-block/src/code-block-utils.ts:250]: I'm not sure but I guess only add `style` to `pre` is good enough?

**Comment by ocavue** [comment]
The code looks good to me. Love your work. I found a weird indentation/padding in the storybook but it's just a smaller issue. I will merge it after the storybook example gets fixed.

File: packages/remirror__extension-code-block/src/code-block-utils.ts (hunk 6 above)
Comment (marcoSven) [reply packages/remirror__extension-code-block/src/code-block-utils.ts:250]: As I mentioned on discord I am trying to avoid touching the code block styling. 

Trying to avoid introducing bugs I will going to revert the changes here.

File: packages/remirror__styles/all.css [hunk 7]
```diff
@@ -2963,6 +3031,10 @@ button:active .remirror-menu-pane-shortcut,
   overflow: auto;
 }
 
+.remirror-pojoaque :has(.remirror-language-select-positioner) ~ pre[class*='language-'] {
+  padding: 2em 1em;
```
Comment (marcoSven) [inline packages/remirror__styles/all.css:3035]: Not sure why some themes use different padding values for the pre element. For example the "pojoaque" theme uses 12px instead of the 1em. 

I think the padding I added to add some spacing for the language select can be adjusted in another pr if needed.

Comment (ocavue) [reply packages/remirror__styles/all.css:3035]: What do you think we don't add padding in `@remirror/styles`, but add some example of CSS code in the readme (and the storybook) to show how to add such padding? I guess whoever wants to use `@remirror/extension-react-language-select` would probably want to add some custom CSS anyway.

Comment (ocavue) [reply packages/remirror__styles/all.css:3035]: I guess maybe some theme re-define `1em`? I'm not sure. I think it's fine since the user might want to add some custom CSS by themselves anyway, so we don't have to make everything perfect.

**Review by ocavue — APPROVED** [review]

**Comment by ocavue** [comment]
@marcoSven Your package has been released https://www.npmjs.com/package/@remirror/extension-react-language-select/v/0.1.0

Thanks again for your contribution.
